pyodata/v2/service.py

```diff
@@ -874,16 +874,10 @@ def nav(self, nav_property):
             association_info.name,
             association_info.namespace)
 
-        navigation_entity_set = None
-        for end in association_set.end_roles:
```

Comment:
why was the for loop removed? Again, all tests are still passing, but this may be a problem of the test suite - that the loop was not covered so far, just simple cases around navigational properties.

Reply:
The loop is not necessary (anymore) because the end_by_role method of the association_set is "looping" over the association_set.end_roles using the generator expression here: https://github.com/SAP/python-pyodata/blob/9a72f930800ed758cf8a4d9c53a8c2434ff49133/pyodata/v2/model.py#L2163

Reply:
OK, that's true. And also, since now test metadata.xml contains multimple association sets ending to same end role (Customer), it is proven by tests.